### PR TITLE
Add actix middleware

### PR DESCRIPTION
**Pull request by qdequele**
This PR fix https://github.com/meilisearch/MeiliSearch/issues/659 and fix https://github.com/meilisearch/MeiliSearch/issues/653.

Two dependencies have been added. They already are in actix-web, at the same version.

### CODE REVIEW

File: meilisearch-http/src/helpers/normalize_slashes.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    fn call(&mut self, mut req: ServiceRequest) -> Self::Future {
+        let head = req.head();
+
+        let path = head.uri.path();
+        let original_len = path.len();
+        let slash_regex = Regex::new("//+").unwrap();
+        let new_path = slash_regex.replace_all(path, "/");
+        let new_path = new_path.trim_end_matches("/");
```
Comment (Kerollmops) [inline meilisearch-http/src/helpers/normalize_slashes.rs:65]: Is it where the url trailing slashes are removed?

Comment (qdequele) [reply meilisearch-http/src/helpers/normalize_slashes.rs:65]: it removes multiples slash and trailing slash.

**Comment by curquiza** [comment]
Could we add tests for that? 🙃
- a slash in the middle
- a slash at the end

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
   - Add support for configuring the lmdb map size (#646, #647)
   - Add exposed port for Dockerfile (#654)
+  - Fix url trailing slash and double slash issues (#659)
+  - Fix accept all Content-Type by default (#653)
+  - Fix returned message on deserialization error
```
Comment (Kerollmops) [inline CHANGELOG.md:7]: Could be more explicit.
```suggestion
  - Return the error message from Serde when a deserialization error is encountered
```

File: meilisearch-http/src/error.rs [hunk 3]
```diff
@@ -180,3 +187,19 @@ impl From<actix_http::Error> for ResponseError {
         ResponseError::Internal(err.to_string())
     }
 }
+
+impl From<JsonPayloadError> for ResponseError {
+    fn from(err: JsonPayloadError) -> ResponseError {
+        match err {
+                JsonPayloadError::Deserialize(err) => ResponseError::BadRequest(format!("BAD_JSON: {}", err)),
```
Comment (Kerollmops) [inline meilisearch-http/src/error.rs:194]:
```suggestion
                JsonPayloadError::Deserialize(err) => ResponseError::BadRequest(format!("Invalid JSON: {}", err)),
```

File: meilisearch-http/src/error.rs [hunk 4]
```diff
@@ -180,3 +187,19 @@ impl From<actix_http::Error> for ResponseError {
...
+                JsonPayloadError::Overflow => ResponseError::PayloadTooLarge,
+                JsonPayloadError::ContentType => ResponseError::UnsupportedMediaType,
+                JsonPayloadError::Payload(err) => ResponseError::BadRequest(format!("Problem decoding request: {}", err)),
```
Comment (Kerollmops) [inline meilisearch-http/src/error.rs:197]:
```suggestion
                JsonPayloadError::Payload(err) => ResponseError::BadRequest(format!("Problem while decoding the request: {}", err)),
```

File: meilisearch-http/src/helpers/normalize_slashes.rs [hunk 5]
```diff
@@ -0,0 +1,87 @@
+///
```
Comment (Kerollmops) [inline meilisearch-http/src/helpers/normalize_slashes.rs:1]: Could you add the original PR address here please?

File: meilisearch-http/src/helpers/normalize_slashes.rs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        let slash_regex = Regex::new("//+").unwrap();
+        let new_path = slash_regex.replace_all(path, "/");
+        let new_path = new_path.trim_end_matches("/");
+
+        if original_len != new_path.len() {
+            let mut parts = head.uri.clone().into_parts();
+
+            let path = match parts.path_and_query.as_ref().map(|pq| pq.query()).flatten() {
+                Some(q) => bytes::Bytes::from(format!("{}?{}", new_path, q)),
+                None =>bytes::Bytes::from(format!("{}", new_path))
```
Comment (Kerollmops) [inline meilisearch-http/src/helpers/normalize_slashes.rs:72]:
```suggestion
                None => bytes::Bytes::from(new_path)
```

File: meilisearch-http/src/lib.rs [hunk 7]
```diff
@@ -28,7 +29,12 @@ pub fn create_app(
 > {
     App::new()
         .app_data(web::Data::new(data.clone()))
-        .app_data(web::JsonConfig::default().limit(1024 * 1024 * 10)) // Json Limit of 10Mb
+        .app_data(
+            web::JsonConfig::default()
+                .limit(1024 * 1024 * 10)
```
Comment (Kerollmops) [inline meilisearch-http/src/lib.rs:34]:
```suggestion
                .limit(1024 * 1024 * 10) // Json Limit of 10Mb
```

File: meilisearch-http/src/helpers/normalize_slashes.rs (hunk 5 above)
Comment (qdequele) [reply meilisearch-http/src/helpers/normalize_slashes.rs:1]: I will not put all the links, so I found that on the issue https://github.com/actix/actix-web/issues/959, and I put the link of gist on the file.

**Pull request merged**